lib/MahoCLI/Commands/PhpstormMetadataGenerate.php

```diff
@@ -0,0 +1,408 @@
+<?php
+
+/**
+ * Maho
+ *
+ * @package    MahoCLI
+ * @copyright  Copyright (c) 2025 Maho (https://mahocommerce.com)
+ * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace MahoCLI\Commands;
+
+use Mage;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use RegexIterator;
+
+#[AsCommand(
+    name: 'phpstorm:metadata:generate',
+    description: 'Generate PhpStorm metadata files for better IDE support',
+)]
+class PhpstormMetadataGenerate extends BaseMahoCommand
+{
+    private array $phpClasses = [];
+
+    private array $moduleConfigs = [
+        'models' => [],
+        'blocks' => [],
+        'helpers' => [],
+        'resourceModels' => [],
+    ];
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->initMaho();
+
+        $metaDir = BP . '/.phpstorm.meta.php';
+        $filesystem = new Filesystem();
+        if ($filesystem->exists($metaDir)) {
+            $filesystem->remove($metaDir);
+        }
+        $filesystem->mkdir($metaDir);
+
+        // Get all .php files
+        $output->writeln('Extracting all PHP classes...');
+        $directoryIterator = new RecursiveDirectoryIterator(
+            MAHO_ROOT_DIR,
+            RecursiveDirectoryIterator::SKIP_DOTS,
+        );
+        $iterator = new RecursiveIteratorIterator(
+            $directoryIterator,
+            RecursiveIteratorIterator::SELF_FIRST,
+        );
+        $phpFiles = new RegexIterator(
+            $iterator,
+            '/^.+\.php$/i',
+            \RecursiveRegexIterator::GET_MATCH,
+        );
+
+        // Extract a list of all classes in those .php files
+        foreach ($phpFiles as $file) {
+            foreach ($this->extractClassesFromFile($file[0]) as $class) {
+                $this->phpClasses[] = $class;
+            }
+        }
+        $output->writeln('<info>Found ' . count($this->phpClasses) . ' classes</info>');
+
+        $this->loadModuleConfigurations($output);
+
+        $this->generateModelsMetadata($output, $metaDir);
+        $this->generateHelpersMetadata($output, $metaDir);
+        $this->generateRegistryMetadata($output, $metaDir);
+        $this->generateBlocksMetadata($output, $metaDir);
+        $this->generateResourceModelsMetadata($output, $metaDir);
+
+        $output->writeln('<info>PhpStorm metadata generation complete!</info>');
+        $output->writeln('<comment>Remember to add .phpstorm.meta.php to your .gitignore file</comment>');
```

Comment:
Do we want this line? We don't currently have `.phpstorm.meta.php` in `.gitignore`.

Reply:
absolutely right, removed